src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/ResourceReferenceResolver.cs

```diff
@@ -74,6 +74,10 @@ public async Task ResolveReferencesAsync(Resource resource, IDictionary<string,
                         {
                             throw new RequestNotValidException(string.Format(Core.Resources.InvalidConditionalReference, reference.Reference));
                         }
+                        else if (results.Count > 1)
```

Comment:
The count needs to be the matched results, so this needs the same filter as the if statement above it.